### DISCUSSION TITLE
verdaccio uplinks problem

### DISCUSSION THREAD

**Discussion by luban-130** [verdaccio/verdaccio#4490] [Q&A]
Verdaccio is deployed without an Internet environment. How to synchronize all packages in the public network npm library to verdaccio?

**Comment by juanpicado** [comment]
This question has been already asked, please search through the issues and discussions.

**Reply by luban-130** [reply]
Is it convenient to attach links to previous discussions?

**Reply by luban-130** [reply]
I deployed verdaccio on a cloud server with a fixed public IP, and then the intranet client pulled the npm package from verdaccio. The uplinks configuration is https://registry.npmmirror.com/, but the intranet client When the client installs an npm package that is not included in verdaccio, an installation failure error will be reported. Does this mean that it is not pulled from uplinks and stored in verdaccio's storage?

**Reply by luban-130** [reply]
for example:
#npm install jsdoc
npm ERR! code E404
npm ERR! 404 Not Found - GET http://172.28.147.27:4873/@jsdoc%2fsalty - no such package available
npm ERR! 404 
npm ERR! 404  '@jsdoc/salty@^0.2.1' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2024-02-22T06_24_24_534Z-debug-0.log

the config.yaml contains uplinks
![image](https://github.com/verdaccio/verdaccio/assets/89766833/a01fbdb2-aca0-4162-9cd0-15ab39ef54b5)
